### PR TITLE
ref(crons): Add test to incident_occurrence module

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is extracted from mark_failed.

Waiting on https://github.com/getsentry/sentry/pull/80792

Part of GH-79328

### CODE REVIEW

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 1]
```diff
@@ -69,83 +61,16 @@ def test_mark_failed_default_params(self, mock_produce_occurrence_to_kafka):
...
-                },
-            },
-        ) == dict(event)
+        assert mock_create_incident_occurrence.call_count == 1
+        assert mock_create_incident_occurrence.call_args == mock.call(
+            checkin,
+            [checkin],
+            monitor_incidents[0],
+            checkin.date_added,
+        )
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:70]: Just validate that we're calling `create_incident_occurrence`

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 2]
```diff
@@ -297,18 +222,7 @@ def test_mark_failed_issue_threshold(self, mock_produce_occurrence_to_kafka):
...
-        kwargs = mock_produce_occurrence_to_kafka.call_args.kwargs
-        occurrence = kwargs["occurrence"]
-        occurrence = occurrence.to_dict()
-        assert occurrence["fingerprint"][0] == monitor_incident.grouphash
-
-        # Human readible failure reason
-        assert (
-            occurrence["evidence_display"][0]["value"]
-            == "3 timeout, 3 missed and 2 error check-ins detected"
-        )
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:311]: Tested in the `incident_occurrences` module now

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 3]
```diff
@@ -297,18 +222,7 @@ def test_mark_failed_issue_threshold(self, mock_produce_occurrence_to_kafka):
         assert monitor_incident.grouphash == monitor_environment.active_incident.grouphash
 
         # assert correct number of occurrences was sent
-        assert len(mock_produce_occurrence_to_kafka.mock_calls) == failure_issue_threshold
-        # assert that the correct uuid fingerprint was sent
-        kwargs = mock_produce_occurrence_to_kafka.call_args.kwargs
-        occurrence = kwargs["occurrence"]
-        occurrence = occurrence.to_dict()
-        assert occurrence["fingerprint"][0] == monitor_incident.grouphash
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:305]: Tested in the `incident_occurrences` module now

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 4]
```diff
@@ -328,12 +242,7 @@ def test_mark_failed_issue_threshold(self, mock_produce_occurrence_to_kafka):
         assert monitor_incident.grouphash == monitor_environment.active_incident.grouphash
 
         # assert correct number of occurrences was sent
-        assert len(mock_produce_occurrence_to_kafka.mock_calls) == failure_issue_threshold + 1
-        # assert that the correct uuid fingerprint was sent
-        kwargs = mock_produce_occurrence_to_kafka.call_args.kwargs
-        occurrence = kwargs["occurrence"]
-        occurrence = occurrence.to_dict()
-        assert occurrence["fingerprint"][0] == monitor_incident.grouphash
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:336]: Tested in the `incident_occurrences` module now

File: tests/sentry/monitors/logic/test_mark_failed.py [hunk 5]
```diff
@@ -432,19 +341,11 @@ def test_mark_failed_issue_threshold_timeout(self, mock_produce_occurrence_to_ka
...
         # assert correct number of occurrences was sent
-        assert len(mock_produce_occurrence_to_kafka.mock_calls) == failure_issue_threshold
-        # assert that the correct uuid fingerprint was sent
-        kwargs = mock_produce_occurrence_to_kafka.call_args.kwargs
-        occurrence = kwargs["occurrence"]
-        occurrence = occurrence.to_dict()
-        assert occurrence["fingerprint"][0] == monitor_incident.grouphash
-
-        # Human readible failure reason
-        assert occurrence["evidence_display"][0]["value"] == "8 timeout check-ins detected"
```
Comment (evanpurkhiser) [inline tests/sentry/monitors/logic/test_mark_failed.py:443]: Tested in the `incident_occurrences` module now

**Review by wedamija — APPROVED** [review]

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**